### DISCUSSION TITLE
do not play piano roll when the music is stopped

### DISCUSSION THREAD

**Discussion by lypwig** [tidalcycles/strudel#586] [Ideas]
The piano roll animation might run when song is stopped.

To reproduce:
- go to the strudel repl;
- without clicking play, click shuffle until there is a song with a piano roll.

The animation runs without the music. If we click play, then stop, the animation is stopped.

That said, I think that the piano roll should not be displayed at all when the song is stopped: it's not really interesting as a static image and it can be disturbing when coding.

**Comment by felixroos** [comment]
similar to https://github.com/tidalcycles/strudel/issues/259

**Comment by lypwig** [comment]
Indeed, sorry for the duplicate.

> That said, I think that the piano roll should not be displayed at all when the song is stopped: it's not really interesting as a static image and it can be disturbing when coding.

What is you though about this?

**Comment by felixroos** [comment]
> That said, I think that the piano roll should not be displayed at all when the song is stopped: it's not really interesting as a static image and it can be disturbing when coding.

Hm I almost never stop the playback so I am coding with the moving image behind anyway.. I also think the image can actually be helpful even when playback is off, especially in the mini repl. There could maybe be a less disturbing mode, where the canvas is placed in the bottom or side bar when activated. This decision also affects other types of visualization, like the (undocumented) animate API + potential future types of visualizations. I think I would maybe find it even more disturbing if the image always blinks in and out of existence when toggling playback.

**Comment by lypwig** [comment]
> I also think the image can actually be helpful even when playback is off

Ok I understand, the piano roll can help to visualize the sounds we create. But apparently it's not possible to update the code without playing it, so we have to press play then stop to update the piano roll.

> I think I would maybe find it even more disturbing if the image always blinks in and out of existence when toggling playback.

Not for me. :) It depends on the user ability to read a text with some shapes on the background.

Maybe add a checkbox somewhere to disable visual effects?

**Comment by felixroos** [comment]
> Maybe add a checkbox somewhere to disable visual effects?

Why can't you just remove `.pianoroll()` ?

> It depends on the user ability to read a text with some shapes on the background.

That's why I was suggesting:

> There could maybe be a less disturbing mode, where the canvas is placed in the bottom or side bar when activated

You can also disable the highlighting via "// strudel disable-highlighting", which could maybe also become a setting.
